GitUI/CommandsDialogs/FormStash.cs

```diff
@@ -187,52 +188,22 @@ private void StashedSelectedIndexChanged(object sender, EventArgs e)
 
             using (WaitCursorScope.Enter())
             {
-                if (stashedItem != null &&
-                    gitStash == _currentWorkingDirStashItem)
+                ObjectId first;
+                ObjectId selected;
+                if (gitStash == _currentWorkingDirStashItem)
                 {
                     // current working directory
-                    View.ViewCurrentChanges(stashedItem, isStaged: false, openWithDifftool: null);
-                }
-                else if (stashedItem != null)
-                {
-                    if (stashedItem.IsNew)
-                    {
-                        View.ViewGitItemAsync(stashedItem);
-                    }
-                    else
-                    {
-                        string extraDiffArguments = View.GetExtraDiffArguments();
-                        Encoding encoding = View.Encoding;
-                        ThreadHelper.JoinableTaskFactory.Run(
-                            () =>
-                            {
-                                Patch patch = Module.GetSingleDiff(gitStash.Name + "^",
-                                    gitStash.Name,
-                                    stashedItem.Name,
-                                    stashedItem.OldName,
-                                    extraDiffArguments,
-                                    encoding,
-                                    true,
-                                    stashedItem.IsTracked);
-                                if (patch == null)
-                                {
-                                    return View.ViewPatchAsync(fileName: null, text: string.Empty, openWithDifftool: null /* not applicable */, isText: true);
-                                }
-
-                                if (stashedItem.IsSubmodule)
-                                {
-                                    return View.ViewPatchAsync(fileName: stashedItem.Name, text: LocalizationHelpers.ProcessSubmodulePatch(Module, stashedItem.Name, patch),
-                                            openWithDifftool: null /* not implemented */, isText: stashedItem.IsSubmodule);
-                                }
-
-                                return View.ViewPatchAsync(fileName: stashedItem.Name, text: patch.Text, openWithDifftool: null /* not implemented */, isText: stashedItem.IsSubmodule);
-                            });
-                    }
+                    first = ObjectId.IndexId;
+                    selected = ObjectId.WorkTreeId;
```

Comment:
> Though I guess stash stores both unstaged and staged files, doesn't it?

Yes
This causes the worktree files to be displayed in the viewer rather than the index, so incorrect...
Will fix the versions
(This is better handled with GitItemStatusWithParent after #7899)

Reply:
worktree and index should be presented in two groups too (this definitely requires #7899)

Reply:
Improved the change to only rev-parse when creating the FileStatusList and separated to own commit - could be a separate PR (but still dependent on this PR).